### PR TITLE
Update FA version to 2.5.6

### PR DESCRIPTION
**Pull request by ksivaman**

### CODE REVIEW

**Review by denera — APPROVED** [review]
LGTM!

File: transformer_engine/pytorch/attention.py [hunk 1]
```diff
@@ -1656,6 +1657,9 @@ def __init__(
         assert (
             _flash_attn_version >= _flash_attn_version_required
         ), f"FlashAttention minimum version {_flash_attn_version_required} is required."
+        assert (
+            _flash_attn_version <= _flash_attn_max_version
+        ), f"FlashAttention maximum version {_flash_attn_max_version} is supported."
```
Comment (ptrendx) [inline transformer_engine/pytorch/attention.py:1662]: Why do this?

Comment (ksivaman) [reply transformer_engine/pytorch/attention.py:1662]: For consistency and adhering to the FA version bounds specified in setup.

**Pull request merged**